### PR TITLE
Fix factory resolver API

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Remove devfile schema validation from factory resolver API.

### What issues does this PR fix or reference?

It is not possible to turn off the schema validation with a query param ```validate=false```.

### How to test this PR?

1. Try to use the swagger and resolve factory with https://gist.githubusercontent.com/olexii4/fc3b757806e660dee0fbe1bc131df3d1/raw/c887e6920f43825755e1569ea5d3ec93ffd9a538/empty.yaml without schema validation. The response code should be 200.

```
{ "url": "https://gist.githubusercontent.com/olexii4/fc3b757806e660dee0fbe1bc131df3d1/raw/c887e6920f43825755e1569ea5d3ec93ffd9a538/empty.yaml" }
```

 Знімок екрана 2023-04-19 о 18 09 28
 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public FactoryMetaDto resolveFactory(
     if (resolvedFactory == null) {
       throw new BadRequestException(FACTORY_NOT_RESOLVABLE);
     }
-    if (validate) {
+    if (validate.equals("true")) {
```
Comment (ibuziuk) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java:115]:
```suggestion
    if ("true".equals(validate)) {
```

**Review by ibuziuk — APPROVED** [review]

**Review by vinokurig — CHANGES_REQUESTED** [review]
The request fails even before the check here:
https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileParser.java#L124
I don't see any problems with the boolean serialisation.

**Comment by olexii4** [comment]
@vinokurig Thanks. I have fixed it.

**Comment by vinokurig** [comment]
@olexii4 I am not sure that removing the validator at all is a right choice. How about fixing the validator itself?

**Pull request closed without merging**